### PR TITLE
6/2 API Changes

### PR DESCRIPTION
**Pull request by IanMatthewHuff**
For #6086

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

**Comment by IanMatthewHuff** [comment]
For some reason my local branch is a bit messed up, it seems to think there was an earlier commit in here. But that was just a previously checked in API submission, so I think this should be fine. I'll just rebase my local after this.

File: src/client/datascience/jupyter/kernels/cellExecution.ts [hunk 1]
```diff
@@ -185,7 +185,8 @@ export class CellExecution {
             `Cell Exec contents ${this.cell.document.getText().substring(0, 50)}...`
         );
         if (!this.canExecuteCell()) {
-            this.execution?.end({});
+            // End state is bool | undefined not optional. Undefined == not success or failure
```
Comment (IanMatthewHuff) [inline src/client/datascience/jupyter/kernels/cellExecution.ts:188]: The new tristate here for success is a bit funky so I called it out with a comment.

File: src/client/datascience/notebook/contentProvider.ts [hunk 2]
```diff
@@ -44,16 +44,15 @@ export class NotebookContentProvider implements VSCNotebookContentProvider {
         if (!this.compatibilitySupport.canOpenWithVSCodeNotebookEditor(uri)) {
             // If not supported, return a notebook with error displayed.
             // We cannot, not display a notebook.
+            const cellData = new NotebookCellData(
```
Comment (IanMatthewHuff) [inline src/client/datascience/notebook/contentProvider.ts:47]: It's funny I thought that typescript had some syntactic sugar for this, like a C# field initializer syntax, but didn't see it.

**Review by joyceerhl — APPROVED** [review]

File: src/client/datascience/notebook/helpers/executionHelpers.ts [hunk 3]
```diff
@@ -48,8 +48,9 @@ export async function updateCellCode(cell: NotebookCell, text: string) {
 export async function addNewCellAfter(cell: NotebookCell, text: string) {
     await chainWithPendingUpdates(cell.notebook, (edit) => {
         traceCellMessage(cell, 'Create new cell after current');
-        edit.replaceNotebookCells(cell.notebook.uri, new NotebookRange(cell.index + 1, cell.index + 1), [
-            new NotebookCellData(NotebookCellKind.Code, text, cell.document.languageId, [], cell.metadata || {})
-        ]);
+        const cellData = new NotebookCellData(NotebookCellKind.Code, text, cell.document.languageId);
+        cellData.outputs = [];
+        cellData.metadata = cell.metadata || {};
```
Comment (DonJayamanne) [inline src/client/datascience/notebook/helpers/executionHelpers.ts:53]:
```suggestion
        cellData.metadata = {};
```

**Review by DonJayamanne — APPROVED** [review]

File: src/client/datascience/notebook/helpers/executionHelpers.ts (hunk 3 above)
Comment (IanMatthewHuff) [reply src/client/datascience/notebook/helpers/executionHelpers.ts:53]: @DonJayamanne Ahh, good catch. This was the same as the old code, but the old code looks like it has a bug. Don't want metadata from the old cell in the new one.

**Review by DavidKutu — APPROVED** [review]

**Comment by IanMatthewHuff** [comment]
Tests look reasonable running manually. Squashing change now. Current failures are expected due to API changes.

**Pull request merged**